tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -505,6 +505,29 @@ def temporary_file_content(self, path, content, binary_mode=True):
     finally:
       os.unlink(path)
 
+  @contextmanager
+  def with_overwritten_file_content(self, file_path, temporary_content=None):
+    """A helper that resets a file after the method runs.
+
+     It will read a file, save the content, maybe write temporary_content to it, yield, then write the
+     original content to the file.
+
+    :param file_path: Absolute path to the file to be reset after the method runs.
+    :param temporary_content: Optional content to write into the file.
+    """
+    with open(file_path, 'r') as f:
+      file_original_content = f.read()
+
+    try:
+      if temporary_content:
```

Comment:
```suggestion
      if temporary_content is not None:
```

Otherwise the empty string won't get written :)